helpdesk-maker-checker-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/makerchecker/MakerCheckerService.java

```diff
@@ -63,58 +74,90 @@ public boolean allChecksPass(SymMessage symMessage) {
   }
 
   /**
-   * Accept a maker checker message.
+   * Approve a maker checker message.
    * Find the message.
    * Send the message to client stream.
    * @param makerCheckerMessage the maker checker message
    */
-  public Set<SymMessage> getAcceptMessages(MakerCheckerMessage makerCheckerMessage) {
-    SymStream stream = new SymStream();
-    stream.setStreamId(makerCheckerMessage.getStreamId());
+  public SymMessage getApprovedMakercheckerMessage(MakerCheckerMessage makerCheckerMessage) {
     try {
-      List<SymMessage> symMessageList = symphonyClient.getMessagesClient()
-          .getMessagesFromStream(stream, makerCheckerMessage.getTimeStamp() - 1, 0, 10);
+      SymStream stream = new SymStream();
+      stream.setStreamId(makerCheckerMessage.getStreamId());
 
-      SymMessage match = null;
-      for(SymMessage symMessage : symMessageList) {
-        if(symMessage.getId().equals(makerCheckerMessage.getMessageId())) {
-          match = symMessage;
-        }
-      }
-
-      if(match == null) {
-        throw new BadRequestException(String.format(MESSAGE_NOT_FOUND, makerCheckerMessage.getMessageId()));
-      }
+      SymMessage symMessage = getApprovedMessage(makerCheckerMessage, stream);
 
-      Set<SymMessage> acceptMessages = new HashSet<>();
+      SymMessage approvedMessage = new SymMessage();
```

Comment:
You have been created a new message and override its data in the loop above. I think you should keep the logic to make approved message inside the checker and return a set of messages gathering the messages generated for each checker.